### PR TITLE
Sun interoperability performance test

### PR DESCRIPTION
**Pull request by Lobo808**
Interoperability performance test

### CODE REVIEW

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+                    .post(url);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (double) (endTime-startTime) / (10^9);
+                cpuUsage[targetIndex] = cpuLoad;
+                freeMemory[targetIndex] = memory;
+                targetIndex++;
+            }
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:96]: issue with time

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+            String body = "{\"id\":\"" + categoryId[j-1] + "\"}";
+            RestAssured.given()
+                    .header("Accept", json)
+                    .body(body)
+                    .post(url);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (endTime-startTime) / (10^9);
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:211]: remove useless double

**Review by MosinLover — CHANGES_REQUESTED** [review]

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+            String body = "{\"id\":\"" + categoryId[j-1] + "\"}";
+            RestAssured.given()
+                    .header("Accept", json)
+                    .body(body)
+                    .post(url);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (double) (endTime-startTime) / (10^9);
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:92]: issue with time, not correct for exponential operator, you have to something like Math.pow(1024, 2) or just 1024 * 1024

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+        // Delete categories from todoitem
+        int targetIndex = targetSize.length-1;
+        for (int j = max; j > 0; j--) {
+            long startTime = System.nanoTime();
+            RestAssured.given().delete(url + "/" + categoryId[j-1]);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (endTime-startTime) / (10^9);
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:154]: ditto

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+        // Delete categories from project
+        int targetIndex = targetSize.length-1;
+        for (int j = max; j > 0; j--) {
+            long startTime = System.nanoTime();
+            RestAssured.given().delete(url + "/" + categoryId[j-1]);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (endTime-startTime) / (10^9);
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:273]: ditto

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+            String body = "{\"id\":\"" + todoId[j-1] + "\"}";
+            RestAssured.given()
+                    .header("Accept", json)
+                    .body(body)
+                    .post(url);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (endTime-startTime) / (10^9);
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:330]: ditto

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+        // Delete todos from category
+        int targetIndex = targetSize.length-1;
+        for (int j = max; j > 0; j--) {
+            long startTime = System.nanoTime();
+            RestAssured.given().delete(url + "/" + todoId[j-1]);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (endTime-startTime) / (10^9);
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:392]: ditto

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+            String body = "{\"id\":\"" + projectId[j-1] + "\"}";
+            RestAssured.given()
+                    .header("Accept", json)
+                    .body(body)
+                    .post(url);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (endTime-startTime) / (10^9);
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:449]: ditto

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java [hunk 9]
```diff
@@ -0,0 +1,530 @@
...
+        // Delete projects from category
+        int targetIndex = targetSize.length-1;
+        for (int j = max; j > 0; j--) {
+            long startTime = System.nanoTime();
+            RestAssured.given().delete(url + "/" + projectId[j-1]);
+            long endTime = System.nanoTime();
+            if (j == targetSize[targetIndex]) {
+                double cpuLoad = osBean.getCpuLoad() * 100;
+                long memory = osBean.getFreeMemorySize() / (1024^2);
+                time[targetIndex] = (double) (endTime-startTime) / (10^9);
```
Comment (MosinLover) [inline TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java:511]: ditto

**Review by MosinLover — CHANGES_REQUESTED** [review]

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java (hunk 2 above)
Comment (Lobo808) [reply TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java]: done

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java (hunk 3 above)
Comment (Lobo808) [reply TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java]: done

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java (hunk 4 above)
Comment (Lobo808) [reply TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java]: done

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java (hunk 5 above)
Comment (Lobo808) [reply TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java]: done

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java (hunk 6 above)
Comment (Lobo808) [reply TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java]: done

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java (hunk 7 above)
Comment (Lobo808) [reply TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java]: done

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java (hunk 8 above)
Comment (Lobo808) [reply TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java]: done

File: TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java (hunk 9 above)
Comment (Lobo808) [reply TodoListManagerTest/src/test/java/PerformanceTest/InteroperabilityPerformanceTest.java]: done

**Comment by MosinLover** [comment]
lgtm

**Pull request merged**